tests/basictest.cpp

```diff
@@ -256,4 +256,216 @@ TEST_F(BasicTest, CompWrongOutputSize) {
+									 "private: y;\n"
+									 "output: z;\n"
+									 "concentrations: {\n"
+									 "y := 20;\n"
+									 "}\n"
+									 "reactions: {\n"
+									 "x -> z;\n"
+									 "y -> z;\n"
+									 "}\n"
+									 "compositions: {\n"
+									 "z = AdditionTwo(y);\n"
+									 "}\n"
+									 "}\n"
+									 "module main {\n"
+									 "private: [a, b, c, d];\n"
+									 "output: e;\n"
+									 "concentrations: {\n"
+									 "a := 50;\n"
+									 "b := 30;\n"
+									 "c := 30;\n"
+									 "}\n"
+									 "compositions: {\n"
+									 "d = Addition(a);\n"
+									 "}\n"
+									 "}\n";
+
+	std::string out = "#!/usr/bin/env crnsimul\n"
+										"main_Addition_0_AdditionTwo_0_y := 20;\n"
+										"main_Addition_0_y := 20;\n"
+										"main_a := 50;\n"
+										"main_b := 30;\n"
+										"main_c := 30;\n"
+										"main_a -> main_d;\n"
+										"main_Addition_0_y -> main_d;\n"
+										"main_Addition_0_y -> main_d;\n"
+										"main_Addition_0_AdditionTwo_0_y -> main_d;\n";
+	driver drv;
+	ASSERT_EQ(drv.parse_string(in), 0);
+	EXPECT_EQ(drv.out, out);
+}
+
+TEST_F(BasicTest, privateSpecSubModReacConc) {
+	std::string in = "module Addition {\n"
+									 "input: x;\n"
+									 "private: y;\n"
+									 "output: z;\n"
+									 "concentrations: {\n"
+									 "y := 20;\n"
+									 "}\n"
+									 "reactions: {\n"
+									 "5x -> z;\n"
+									 "10y -> z;\n"
+									 "}\n"
+									 "}\n"
+									 "module main {\n"
+									 "private: [a, b, c, d];\n"
+									 "output: e;\n"
+									 "concentrations: {\n"
+									 "a := 50;\n"
+									 "b := 30;\n"
+									 "c := 30;\n"
+									 "}\n"
+									 "compositions: {\n"
+									 "d = Addition(a);\n"
+									 "}\n"
+									 "}\n";
+
+	std::string out = "#!/usr/bin/env crnsimul\n"
+										"main_Addition_0_y := 20;\n"
+										"main_a := 50;\n"
+										"main_b := 30;\n"
+										"main_c := 30;\n"
+										"5main_a -> main_d;\n"
+										"10main_Addition_0_y -> main_d;\n";
+	driver drv;
+	ASSERT_EQ(drv.parse_string(in), 0);
+	EXPECT_EQ(drv.out, out);
+}
+
+TEST_F(BasicTest, outputSpecInSubModConc) {
+	std::string in = "module Addition {\n"
+									 "input: x;\n"
+									 "private: y;\n"
+									 "output: z;\n"
+									 "concentrations: {\n"
+									 "y := 20;\n"
+									 "z := 20;\n"
+									 "}\n"
+									 "reactions: {\n"
+									 "5x -> z;\n"
+									 "10y -> z;\n"
+									 "}\n"
+									 "}\n"
+									 "module main {\n"
+									 "private: [a, b, c, d];\n"
+									 "output: e;\n"
+									 "concentrations: {\n"
+									 "a := 50;\n"
+									 "b := 30;\n"
+									 "c := 30;\n"
+									 "}\n"
+									 "compositions: {\n"
+									 "d = Addition(a);\n"
+									 "}\n"
+									 "}\n";
+
+	std::string out = "#!/usr/bin/env crnsimul\n"
+										"main_Addition_0_y := 20;\n"
+										"main_Addition_0_z := 20;\n"
+										"main_a := 50;\n"
```

Comment:
Men z skal jo bare være et alias for d? Så i stedet for `main_Addition_0_z := 20;` burde der stå `main_d := 20`.

Reply:
Nu tog jeg bare der gamle der stod, så det virker til output specie, men skal nok få fikset det.